memory/PacketPoolAllocator.h

```diff
@@ -50,4 +50,35 @@ inline Packet* makePacket(PacketPoolAllocator& allocator, const Packet& packet)
     return makePacket(allocator, packet.get(), packet.getLength());
 }
 
+// Be very careful with reset as the deleter is not changed if already set. You may try to deallocate
+// packet in the wrong pool.
+typedef std::unique_ptr<memory::Packet, PacketPoolAllocator::Deleter> PacketPtr;
```

Comment:
I was wondering if the alias shouldn't be something with "unique" (example: UniquePacket)